draft-ietf-quic-http.md

```diff
@@ -492,16 +492,28 @@ this header is determined by the stream type.
 ~~~~~~~~~~
 {: #fig-stream-header title="Unidirectional Stream Header"}
 
-Two stream types are defined in this document: control streams
-({{control-streams}}) and push streams ({{server-push}}).  Other stream types
-can be defined by extensions to HTTP/QUIC.
+Some stream types are reserved ({{stream-grease}}).  Two stream types are
+defined in this document: control streams ({{control-streams}}) and push streams
+({{server-push}}).  Other stream types can be defined by extensions to
+HTTP/QUIC.
 
 If the stream header indicates a stream type which is not supported by the
-recipient, this SHOULD be treated as a stream error of type
-HTTP_UNKNOWN_STREAM_TYPE.  The semantics of the remainder of the stream are
+recipient, the remainder of the stream cannot be consumed as the semantics are
 unknown. Implementations SHOULD NOT send stream types the peer is not already
-known to support, since a stream error can be promoted to a connection error at
-the peer's discretion (see {{errors}}).
+known to support. Recipients of unknown stream types MAY trigger a QUIC
+STOP_SENDING frame with an error code of HTTP_UNKNOWN_STREAM_TYPE, but MUST NOT
+consider such streams to be an error of any kind.
+
+### Reserved Stream Types {#stream-grease}
+
+Stream types of the format `0x1f * N` are reserved to exercise the requirement
+that unknown types be ignored. These streams have no semantic meaning, and can
+be sent when application-layer padding is desired.  They MAY also be sent on
```

Comment:
What do you mean by application-layer padding in this case? If this is some property of the mapping that can be used to add additional padding to QUIC's PADDING frames then is it worth pointing out in security considerations?

Reply:
Perhaps.  HTTP/2 allowed padding on the HEADERS and DATA frames, to make analysis of the traffic more difficult.  QUIC can pad individual packets to different / uniform sizes, but I'm assuming that most stacks won't generate traffic where none exists.  An application layer can make analysis more difficult by sending traffic even when no requests are happening.

Reply:
Ok so I get the problem, this is more like a background level of noise.

I guess I had been expecting transport libraries to handle packing and padding with some influence from the application. It wouldn't be a stretch for them to provide an API that would allow the caller to declare a background level of traffic, however the transport won't have any stream to send this on.

In that case, I like the side effect of this.

Reply:
Added a mention in Security Considerations.

Reply:
Padding using bogus frames might work, but there are two problems with this:

1. Not all streams use frames.
2. Padding that uses bogus unidirectional streams isn't guaranteed to be placed in the same packet as the data that it pads.